aws-android-sdk-core/src/main/java/com/amazonaws/util/json/JsonUtils.java

```diff
@@ -141,6 +141,87 @@ public static AwsJsonWriter getJsonWriter(Writer out) {
+                String key = reader.nextName();
+                if (reader.isContainer()) {
+                    if (AwsJsonToken.BEGIN_ARRAY.equals(reader.peek())) {
+                        final StringWriter out = new StringWriter();
+                        final AwsJsonWriter writer = getJsonWriter(out);
+                        reader.beginArray();
+                        writer.beginArray();
+                        try {
+                            while (!AwsJsonToken.END_ARRAY.equals(reader.peek())) {
+                                AwsJsonToken nextToken = reader.peek();
+                                if (AwsJsonToken.BEGIN_OBJECT.equals(nextToken)){
+                                    reader.beginObject();
+                                    writer.beginObject();
+                                } else if (AwsJsonToken.FIELD_NAME.equals(nextToken)){
+                                    String name = reader.nextName();
+                                    // do not write key if the value is a nested array
+                                    if (!AwsJsonToken.BEGIN_ARRAY.equals(reader.peek())) {
+                                        writer.name(name);
+                                    }
+                                } else if (AwsJsonToken.END_OBJECT.equals(nextToken)){
+                                    reader.endObject();
+                                    writer.endObject();
+                                } else if (AwsJsonToken.END_ARRAY.equals(nextToken)) {
+                                    reader.endArray();
+                                    writer.endArray();
+                                } else if (AwsJsonToken.VALUE_STRING.equals(nextToken)
+                                        || AwsJsonToken.VALUE_NUMBER.equals(nextToken)
+                                        || AwsJsonToken.VALUE_NULL.equals(nextToken)
+                                        || AwsJsonToken.VALUE_BOOLEAN.equals(nextToken)) {
+                                    String value = reader.nextString();
+                                    writer.value(value);
+                                }
+                                else {
+                                    // skip non string nodes inside the array
+                                    reader.skipValue();
+                                }
+                            }
+                        } catch (IOException e) {
+                            e.printStackTrace();
+                        }
+                        reader.endArray();
+                        writer.endArray();
+                        writer.flush();
+                        writer.close();
+                        map.put(key, out.toString());
+                    } else {
+                        // skip non string or non array nodes
+                        reader.skipValue();
+                    }
+                } else {
+                    map.put(key, reader.nextString());
```

Comment:
maybe add a comment here to say if its not an array it falls into this else and just gets mapped like normal